scalarxmat44_neon.c

```diff
@@ -0,0 +1,129 @@
+#include <arm_neon.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+
+#define N 4
+#define LOOPS 1000000000
+
+void scalarxmat44_c(float *A[], float lamda) {
+  for (int i = 0; i < 4; i++) {
+    for (int j = 0; j < 4; j++) {
+      A[i][j] = lamda * A[i][j];
+    }
+  }
+}
+
+void scalarxmat44_neon(float32x4_t lamda_vector, float32x4_t array_in,
+                       float *B[]) {
+  for (int i = 0; i < 4; i++) {
+    // load 4 float values from the array B, multiply each value with lamda
+    // and then store result back to B array
+    array_in = vld1q_f32(&B[i][0]);
```

Comment:
you are not using the passed argument array_in, you could just declare it directly here.

Reply:
and remove the function parameter.

Reply:
didn't do it this way because it would get declared N times, but this way it's only declared once